File: src/main/java/me/mrlion/lionplugin/commands/ReloadCommand.java
package me.mrlion.lionplugin.commands;

import me.mrlion.lionplugin.LionPlugin;
import me.mrlion.lionplugin.util.ColorUtils;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.jetbrains.annotations.NotNull;

import java.util.Objects;
import java.util.logging.Level;

public class ReloadCommand implements CommandExecutor {
    private final LionPlugin plugin;

    public ReloadCommand(LionPlugin plugin) {
        this.plugin = plugin;
    }

    @Override
    public boolean onCommand(CommandSender sender, @NotNull Command command, @NotNull String label, String[] args) {
        if (!sender.hasPermission("lionplugin.reload")) {
            String reloadNoPerms = Objects.requireNonNull(plugin.getConfig().getString("reload.no-permissions"));
            sender.sendMessage(ColorUtils.colorize(reloadNoPerms));
            return true;
        }

        try {
            String reloadMessage = Objects.requireNonNull(plugin.getConfig().getString("reload.config-reloaded"));
            sender.sendMessage(ColorUtils.colorize(reloadMessage));
            plugin.reloadConfig();
        } catch (Exception e) {
            String reloadError = Objects.requireNonNull(plugin.getConfig().getString("reload.error"));
            sender.sendMessage(ColorUtils.colorize(reloadError));
            plugin.getLogger().log(Level.SEVERE, "Error occurred while reloading the config:", e);
        }
        return true;
    }
}

File: src/main/java/me/mrlion/lionplugin/commands/ReloadCommand.kt
package me.mrlion.lionplugin.commands

import me.mrlion.lionplugin.LionPlugin
import org.bukkit.ChatColor
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import java.util.*

class ReloadCommand(private val plugin: LionPlugin) : CommandExecutor {
    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<String>): Boolean {
        if (!sender.hasPermission("lionplugin.reload")) {
            val noPermsMessage = Objects.requireNonNull(plugin.config.getString("other.no-perms"))?.let { ChatColor.translateAlternateColorCodes('&', it) }
            if (noPermsMessage != null) {
                if (noPermsMessage.isEmpty()) {
                    sender.sendMessage(ChatColor.RED.toString() + "You don't have permission to use this command.")
                } else {
                    sender.sendMessage(noPermsMessage)
                }
            }
            return true
        }

        plugin.reloadConfig()
        val configReloadedMessage = Objects.requireNonNull(plugin.config.getString("reload.config-reloaded"))?.let { ChatColor.translateAlternateColorCodes('&', it) }
        sender.sendMessage(configReloadedMessage)
        plugin.logger.info("Config was successfully reloaded")
        return true
    }
}

